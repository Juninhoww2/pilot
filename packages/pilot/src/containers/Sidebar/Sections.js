import React from 'react'
import PropTypes from 'prop-types'
import {
  propOr,
  __,
} from 'ramda'

import SidebarSections from '../../components/SidebarSections'
import isPaymentLink from '../../validation/isPaymentLink'

import formatDecimalCurrency from '../../formatters/decimalCurrency'

const MINIMUM_API_VALUE = 100

const Sections = ({
  balance,
  companyType,
  onWithdraw,
  t,
  transfersPricing,
}) => {
  const getFrombalance = propOr(null, __, balance)
  const available = getFrombalance('available')

  const minimumWithdrawalValue = transfersPricing.ted + MINIMUM_API_VALUE
  const isCompanyNotPaymentLink = !!companyType && !isPaymentLink(companyType)

  return (
    <SidebarSections
      sections={[
        {
          action: onWithdraw,
          actionTitle: t('pages.sidebar.withdraw'),
          disabled: available <= minimumWithdrawalValue,
          showButton: isCompanyNotPaymentLink,
          title: t('pages.sidebar.available'),
          value: <span><small>{t('pages.sidebar.currency_symbol')}</small> {formatDecimalCurrency(available)}</span>,
        },
      ]}
    />
  )
}

Sections.propTypes = {
  balance: PropTypes.shape({
    available: PropTypes.number,
  }).isRequired,
  companyType: PropTypes.string,
  onWithdraw: PropTypes.func,
  t: PropTypes.func.isRequired,
  transfersPricing: PropTypes.shape({
    ted: PropTypes.number,
  }),
}

Sections.defaultProps = {
  companyType: '',
  onWithdraw: null,
  transfersPricing: {},
}

export default Sections
