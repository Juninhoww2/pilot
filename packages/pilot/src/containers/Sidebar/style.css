@import "former-kit-skin-pagarme/dist/styles/colors/light.css";
@import "former-kit-skin-pagarme/dist/styles/spacing.css";

.sidebarIcon {
  fill: var(--color-squanchy-gray-100);
}
