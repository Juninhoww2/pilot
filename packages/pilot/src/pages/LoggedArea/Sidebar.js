import React from 'react'
import PropTypes from 'prop-types'
import {
  curry,
  head,
  isNil,
  pipe,
  split,
} from 'ramda'

import {
  matchPath,
  withRouter,
} from 'react-router-dom'

import SidebarContainer from '../../containers/Sidebar'
import Logo from '../../components/Logo'

import isLinkmeSeller from '../../validation/isLinkmeSeller'

const removeRouteParams = pipe(
  split(':'),
  head
)

const handleLinkClick = curry((push, currentPath, route) => {
  const matched = matchPath(currentPath, route)

  if (isNil(matched) || !route.active) {
    push(route.path)
  }
})

const Sidebar = ({
  // This block of code is commented because of issue #1159 (https://github.com/pagarme/pilot/issues/1159)
  // It was commented on to remove the anticipation limits call on Balance page
  // This code will be used again in the future when ATLAS project implements the anticipation flow
  // More details in issue #1159
  // anticipationLimit,
  balance,
  companyName,
  companyType,
  history,
  location: { pathname },
  recipientId,
  routes,
  sessionId,
  t,
  transfersPricing,
  user,
}) => (
  <SidebarContainer
    balance={balance}
    companyName={companyName}
    companyType={companyType}
    showBalance={!isLinkmeSeller({
      company: {
        type: companyType,
      },
      user,
    })}
    links={routes
      .filter(({ hidden }) => !hidden)
      .map(route => ({
        ...route,
        active: pathname.includes(removeRouteParams(route.path)),
      }))
    }
    logo={Logo}
    // This block of code is commented because of issue #1159 (https://github.com/pagarme/pilot/issues/1159)
    // It was commented on to remove the anticipation limits call on Balance page
    // This code will be used again in the future when ATLAS project implements the anticipation flow
    // More details in issue #1159
    // anticipationLimit={anticipationLimit}
    onLinkClick={handleLinkClick(history.push, pathname)}
    onWithdraw={() => history.push(`/withdraw/${recipientId}`)}
    sessionId={sessionId}
    t={t}
    transfersPricing={transfersPricing}
  />
)

Sidebar.propTypes = {
  // This block of code is commented because of issue #1159 (https://github.com/pagarme/pilot/issues/1159)
  // It was commented on to remove the anticipation limits call on Balance page
  // This code will be used again in the future when ATLAS project implements the anticipation flow
  // More details in issue #1159
  // anticipationLimit: PropTypes.number,
  balance: PropTypes.shape({
    available: PropTypes.number,
  }).isRequired,
  companyName: PropTypes.string,
  companyType: PropTypes.string,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }).isRequired,
  recipientId: PropTypes.string,
  routes: PropTypes.arrayOf(
    PropTypes.shape({
      hidden: PropTypes.bool,
      path: PropTypes.string,
    })
  ).isRequired,
  sessionId: PropTypes.string,
  t: PropTypes.func.isRequired,
  transfersPricing: PropTypes.shape({
    ted: PropTypes.number,
  }),
  user: PropTypes.shape({
    permission: PropTypes.string,
  }),
}

Sidebar.defaultProps = {
  // This block of code is commented because of issue #1159 (https://github.com/pagarme/pilot/issues/1159)
  // It was commented on to remove the anticipation limits call on Balance page
  // This code will be used again in the future when ATLAS project implements the anticipation flow
  // More details in issue #1159
  // anticipationLimit: null,
  companyName: '',
  companyType: '',
  recipientId: null,
  sessionId: '',
  transfersPricing: {},
  user: {},
}

export default withRouter(Sidebar)
