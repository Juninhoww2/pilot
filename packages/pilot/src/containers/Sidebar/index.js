import React, { isValidElement } from 'react'
import PropTypes from 'prop-types'
import {
  propOr,
  __,
  ifElse,
  equals,
  always,
} from 'ramda'

import {
  Button,
  Flexbox,
  Popover,
  PopoverContent,
  Sidebar,
  SidebarHeader,
  SidebarLink,
  SidebarLinks,
} from 'former-kit'

import IconMenuCollapsed from 'emblematic-icons/svg/MenuCollapsed24.svg'
import IconMenuNotCollapsed from 'emblematic-icons/svg/MenuNotCollapsed24.svg'

import IconWallet from 'emblematic-icons/svg/Wallet32.svg'

import style from './style.css'

import SidebarSections from '../../components/SidebarSections'
import SidebarSummary from '../../components/SidebarSummary'
import formatDecimalCurrency from '../../formatters/decimalCurrency'
import environment from '../../environment'

const MINIMUM_API_VALUE = 100

const getMenuIcon = ifElse(
  equals(true),
  always(IconMenuCollapsed),
  always(IconMenuNotCollapsed)
)

class SidebarContainer extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      collapsed: false,
      summaryCollapsed: false,
    }

    this.handleToggleSidebar = this.handleToggleSidebar.bind(this)
  }

  handleToggleSidebar () {
    const { collapsed } = this.state
    this.setState({ collapsed: !collapsed })
  }

  renderSections () {
    const {
      // This block of code is commented because of issue #1159 (https://github.com/pagarme/pilot/issues/1159)
      // It was commented on to remove the anticipation limits call on Balance page
      // This code will be used again in the future when ATLAS project implements the anticipation flow
      // More details in issue #1159
      // anticipationLimit,
      balance,
      onWithdraw,
      t,
      transfersPricing,
    } = this.props

    const getFrombalance = propOr(null, __, balance)
    const available = getFrombalance('available')

    const minimumWithdrawalValue = transfersPricing.ted + MINIMUM_API_VALUE

    return (
      <SidebarSections
        sections={[
          {
            action: onWithdraw,
            actionTitle: t('pages.sidebar.withdraw'),
            disabled: available <= minimumWithdrawalValue,
            title: t('pages.sidebar.available'),
            value: <span><small>{t('pages.sidebar.currency_symbol')}</small> {formatDecimalCurrency(available)}</span>,
          },
        ]}
      />
    )
  }

  render () {
    const {
      collapsed,
      summaryCollapsed,
    } = this.state
    const {
      companyName,
      companyType,
      links,
      logo: Logo,
      onLinkClick,
      sessionId,
      t,
    } = this.props

    const MenuIcon = getMenuIcon(collapsed)

    return (
      <Sidebar collapsed={collapsed}>
        <SidebarHeader>
          {!collapsed && <Logo width={140} />}

          <Button
            onClick={this.handleToggleSidebar}
            icon={(
              <MenuIcon
                width={24}
                height={24}
                className={style.sidebarIcon}
              />
            )}
            fill="clean"
            relevance="low"
          />
        </SidebarHeader>

        {!collapsed
          && (
            <SidebarSummary
              collapsed={summaryCollapsed}
              onClick={() => this.setState({
                summaryCollapsed: !summaryCollapsed,
              })}
              subtitle={
                summaryCollapsed
                  ? t('pages.sidebar.show_balance')
                  : t('pages.sidebar.hide_balance')
              }
              title={companyName}
            >
              {this.renderSections()}
            </SidebarSummary>
          )
        }

        {collapsed
          && (
            <Popover
              base="dark"
              content={(
                <PopoverContent>
                  {this.renderSections()}
                </PopoverContent>
              )}
              placement="rightStart"
            >
              <SidebarLinks>
                <SidebarLink
                  collapsed={collapsed}
                  onClick={() => null}
                  icon={<IconWallet width={16} height={16} />}
                  title={t('pages.sidebar.balance')}
                />
              </SidebarLinks>
            </Popover>
          )
        }

        <SidebarLinks>
          {links.map(({
            active,
            exact,
            icon: Icon,
            path,
            title,
          }) => (
            <SidebarLink
              key={path}
              title={t(title)}
              active={active}
              icon={<Icon width={16} height={16} />}
              collapsed={collapsed}
              onClick={() => onLinkClick({ active, exact, path })}
            />
          ))}
        </SidebarLinks>
        {!collapsed && companyType !== 'payment_link_app'
          && (
            <Flexbox
              className={style.backToOldVersion}
              justifyContent="center"
            >
              <Button
                onClick={
                  // eslint-disable-next-line no-undef
                  () => window.open(`https://dashboard.pagar.me/#login?session_id=${sessionId}&redirect_to=dashboard.home&environment=${environment}`)
                }
                fill="outline"
                size="tiny"
              >
                {t('pages.sidebar.back_to_old_version')}
              </Button>
            </Flexbox>
          )
        }
      </Sidebar>
    )
  }
}

SidebarContainer.propTypes = {
  // This block of code is commented because of issue #1159 (https://github.com/pagarme/pilot/issues/1159)
  // It was commented on to remove the anticipation limits call on Balance page
  // This code will be used again in the future when ATLAS project implements the anticipation flow
  // More details in issue #1159
  // anticipationLimit: PropTypes.number,
  balance: PropTypes.shape({
    available: PropTypes.number,
    waitingFunds: PropTypes.number,
  }).isRequired,
  companyName: PropTypes.string,
  companyType: PropTypes.string,
  links: PropTypes.arrayOf(PropTypes.shape({
    active: PropTypes.bool,
    component: isValidElement,
    icon: PropTypes.func,
    path: PropTypes.string,
    title: PropTypes.string,
  })).isRequired,
  logo: PropTypes.func.isRequired,
  onLinkClick: PropTypes.func.isRequired,
  onWithdraw: PropTypes.func,
  sessionId: PropTypes.string,
  t: PropTypes.func.isRequired,
  transfersPricing: PropTypes.shape({
    ted: PropTypes.number,
  }),
}

SidebarContainer.defaultProps = {
  // This block of code is commented because of issue #1159 (https://github.com/pagarme/pilot/issues/1159)
  // It was commented on to remove the anticipation limits call on Balance page
  // This code will be used again in the future when ATLAS project implements the anticipation flow
  // More details in issue #1159
  // anticipationLimit: null,
  companyName: '',
  companyType: '',
  onWithdraw: null,
  sessionId: '',
  transfersPricing: {},
}

export default SidebarContainer
