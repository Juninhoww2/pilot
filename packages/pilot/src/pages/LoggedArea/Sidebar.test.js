import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import Sidebar from './Sidebar'

test('should not show recipient tab when companyCanCreateRecipient is false', () => {
  const { container, queryByText } = render(
    <Sidebar
      balance={{
        available: 10,
      }}
      companyCanCreateRecipient={false}
      t={v => v}
    />,
    { wrapper: MemoryRouter }
  )

  expect(queryByText(/pages.recipients.title/i)).not.toBeInTheDocument()
  expect(container).toMatchSnapshot()
})

test('should show recipient tab when companyCanCreateRecipient is true', () => {
  const { container, getByText } = render(
    <Sidebar
      balance={{
        available: 10,
      }}
      companyCanCreateRecipient
      t={v => v}
    />,
    { wrapper: MemoryRouter }
  )

  getByText(/pages.recipients.title/i)
  expect(container).toMatchSnapshot()
})

test('should show recipient tab when companyType is paymentLink', () => {
  const { container, queryByText } = render(
    <Sidebar
      balance={{
        available: 10,
      }}
      companyCanCreateRecipient
      companyType="payment_link_app"
      t={v => v}
    />,
    { wrapper: MemoryRouter }
  )

  expect(queryByText(/pages.recipients.title/i)).not.toBeInTheDocument()
  expect(container).toMatchSnapshot()
})

