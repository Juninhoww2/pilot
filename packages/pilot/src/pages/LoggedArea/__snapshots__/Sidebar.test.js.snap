// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not show recipient tab when companyCanCreateRecipient is false 1`] = `
<div>
  <aside
    class="undefined"
  >
    <header>
      <svg
        data-file-name="SvgLogoLive"
        width="140"
      />
      <button
        class=""
        type="button"
      >
        <svg
          class="sidebarIcon"
          data-file-name="SvgMenuNotCollapsed24"
          height="24"
          width="24"
        />
        <span
          style="height: 0px; left: 0px; top: 0px; width: 0px;"
        />
      </button>
    </header>
    <div
      class="summary expanded"
    >
      <button
        class="title"
        role="link"
        type="button"
      >
        
        <span
          class="subtitle"
        >
          pages.sidebar.hide_balance
          <div
            class="undefined"
          />
          <svg
            data-file-name="SvgChevronUp32"
            height="16"
            width="16"
          />
        </span>
      </button>
      <div
        class="sections"
      >
        <ul>
          <li
            class="item"
          >
            <span
              class="title"
            >
              pages.sidebar.available
            </span>
            <div
              class="value"
            >
              <span>
                <small>
                  pages.sidebar.currency_symbol
                </small>
                 
                0,10
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <nav>
      <ul>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgHome32"
                  height="16"
                  width="16"
                />
              </span>
              pages.home.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgHeartRate32"
                  height="16"
                  width="16"
                />
              </span>
              pages.balance.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgCheckout32"
                  height="16"
                  width="16"
                />
              </span>
              pages.transactions.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgLink32"
                  height="16"
                  width="16"
                />
              </span>
              pages.payment_links.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgWrench32"
                  height="16"
                  width="16"
                />
              </span>
              pages.settings.company.menu
            </div>
          </button>
        </li>
      </ul>
    </nav>
    <div
      class="backToOldVersion"
    >
      <button
        class=""
        type="button"
      >
        <span>
          pages.sidebar.back_to_old_version
        </span>
        <span
          style="height: 0px; left: 0px; top: 0px; width: 0px;"
        />
      </button>
    </div>
  </aside>
</div>
`;

exports[`should show recipient tab when companyCanCreateRecipient is true 1`] = `
<div>
  <aside
    class="undefined"
  >
    <header>
      <svg
        data-file-name="SvgLogoLive"
        width="140"
      />
      <button
        class=""
        type="button"
      >
        <svg
          class="sidebarIcon"
          data-file-name="SvgMenuNotCollapsed24"
          height="24"
          width="24"
        />
        <span
          style="height: 0px; left: 0px; top: 0px; width: 0px;"
        />
      </button>
    </header>
    <div
      class="summary expanded"
    >
      <button
        class="title"
        role="link"
        type="button"
      >
        
        <span
          class="subtitle"
        >
          pages.sidebar.hide_balance
          <div
            class="undefined"
          />
          <svg
            data-file-name="SvgChevronUp32"
            height="16"
            width="16"
          />
        </span>
      </button>
      <div
        class="sections"
      >
        <ul>
          <li
            class="item"
          >
            <span
              class="title"
            >
              pages.sidebar.available
            </span>
            <div
              class="value"
            >
              <span>
                <small>
                  pages.sidebar.currency_symbol
                </small>
                 
                0,10
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <nav>
      <ul>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgHome32"
                  height="16"
                  width="16"
                />
              </span>
              pages.home.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgHeartRate32"
                  height="16"
                  width="16"
                />
              </span>
              pages.balance.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgCheckout32"
                  height="16"
                  width="16"
                />
              </span>
              pages.transactions.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgLink32"
                  height="16"
                  width="16"
                />
              </span>
              pages.payment_links.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgAnticipation32"
                  height="16"
                  width="16"
                />
              </span>
              pages.recipients.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgWrench32"
                  height="16"
                  width="16"
                />
              </span>
              pages.settings.company.menu
            </div>
          </button>
        </li>
      </ul>
    </nav>
    <div
      class="backToOldVersion"
    >
      <button
        class=""
        type="button"
      >
        <span>
          pages.sidebar.back_to_old_version
        </span>
        <span
          style="height: 0px; left: 0px; top: 0px; width: 0px;"
        />
      </button>
    </div>
  </aside>
</div>
`;

exports[`should show recipient tab when companyType is paymentLink 1`] = `
<div>
  <aside
    class="undefined"
  >
    <header>
      <svg
        data-file-name="SvgLogoLive"
        width="140"
      />
      <button
        class=""
        type="button"
      >
        <svg
          class="sidebarIcon"
          data-file-name="SvgMenuNotCollapsed24"
          height="24"
          width="24"
        />
        <span
          style="height: 0px; left: 0px; top: 0px; width: 0px;"
        />
      </button>
    </header>
    <div
      class="summary expanded"
    >
      <button
        class="title"
        role="link"
        type="button"
      >
        
        <span
          class="subtitle"
        >
          pages.sidebar.hide_balance
          <div
            class="undefined"
          />
          <svg
            data-file-name="SvgChevronUp32"
            height="16"
            width="16"
          />
        </span>
      </button>
      <div
        class="sections"
      >
        <ul>
          <li
            class="item"
          >
            <span
              class="title"
            >
              pages.sidebar.available
            </span>
            <div
              class="value"
            >
              <span>
                <small>
                  pages.sidebar.currency_symbol
                </small>
                 
                0,10
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <nav>
      <ul>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgHome32"
                  height="16"
                  width="16"
                />
              </span>
              pages.home.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgHeartRate32"
                  height="16"
                  width="16"
                />
              </span>
              pages.balance.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgCheckout32"
                  height="16"
                  width="16"
                />
              </span>
              pages.transactions.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgLink32"
                  height="16"
                  width="16"
                />
              </span>
              pages.payment_links.title
            </div>
          </button>
        </li>
        <li
          class=""
        >
          <button
            role="link"
            type="button"
          >
            <div>
              <span>
                <svg
                  data-file-name="SvgWrench32"
                  height="16"
                  width="16"
                />
              </span>
              pages.settings.company.menu
            </div>
          </button>
        </li>
      </ul>
    </nav>
  </aside>
</div>
`;
